### ISSUE TITLE
Server block - Query error

### ISSUE THREAD

**Issue by SmokePuddingEveryday** [montoyo/webdisplays#81]
Trying to use a server block on 1.12.2 WebDisplays. Upon trying to use most commands (ls, quota, url), I get a 'Query Error, try "reconnect".' message. Am I doing something wrong here?

https://i.imgur.com/q7DgqF3.png

**Comment by ItsK1tty** [comment]
@montoyo

**Comment by montoyo** [comment]
Interesting; will most definitely need your logs. If you don't know where to find them, check [this](https://montoyo.net/wdwiki/index.php?title=Main_Page#Where_are_my_log_files.3F) out.
Also try the reconnect command as suggested and screenshot the output here.

**Comment by ItsK1tty** [comment]
oh, you actually answered! I thought, this repo is no longer maintained and I'm the only active person left

**Comment by montoyo** [comment]
Yeah. Since Nic got MCEF working on Mac I figured that maybe MCEF and WebDisplays do have a future. Guess we'll see.
Also that issue is interesting.

**Comment by SmokePuddingEveryday** [comment]
WebDisplays Version 1.1 (Minecraft 1.12.2)
Forge Server 14.23.5.2838

How to reproduce: Place down Server block, right click, and attempt to use any of the commands "ls", "quota", and "url"

Video: https://streamable.com/0in8zm
Server logs: https://hastebin.com/oqetayesot.md

The issue does not occur in singleplayer. Let me know if I'm missing any info.

**Comment by montoyo** [comment]
Thank you. For some reason I don't see any WebDisplays/Miniserv-related information in the server logs, which is very weird... could you also send us the WebDisplays config of the server?

Also, are you familiar with the concept of ports? MiniServ (= the server block) uses an extra port, 25566 by default. If another process is using this port or if the server's firewall is blocking it, the server block will not work.

**Comment by ItsK1tty** [comment]
why wouldn't WD check if its working port is already binded and use some other one, say, 25567

**Comment by montoyo** [comment]
Because I forgot or I was lazy, can't remember. But yeah, that's something that should be added.
Solves only part of the problem though: it will make sure the port is unbound but I don't think there's a way to check if the port is blocked by the firewall.

**Comment by SmokePuddingEveryday** [comment]
> Also, are you familiar with the concept of ports? MiniServ (= the server block) uses an extra port, 25566 by default. If another process is using this port or if the server's firewall is blocking it, the server block will not work.

https://hastebin.com/wemonojoje.php

I'm familiar with ports, however, I'm hosting the server on server.pro, so I'm not sure if I'd be able to access the firewall, which is probably the problem.

**Comment by ItsK1tty** [comment]
try contacting their support

**Comment by montoyo** [comment]
Thanks for sending the config. So yeah, miniServ is enabled on port 25566. Before contacting the support, you could try to change the `miniservPort` property in the config file on the server **and all clients**

**Comment by ItsK1tty** [comment]
I don't think that would go, though: if I was an admin, I would configure the firewall that way, so only 25565 will stay open, so whatever they will try to change it to, it will still be open. That's just a theory, though..

**Comment by montoyo** [comment]
Maybe, maybe not... I think rcon is using 25566 by default and that's something a commercial server would have enabled...

In any case, its worth trying

**Comment by ItsK1tty** [comment]
probably..
~~also, montoyo, take a look on #78 and #82, these issues might be also "interesting"~~

**Comment by montoyo** [comment]
Nah. This is most certainly due to mod incompatibility. Also the guy kept mentioning me for no goddamn reason.

**Comment by SmokePuddingEveryday** [comment]
Thanks for the suggestion. I'll try changing the port next time I'm the only one on the server.

**Comment by SmokePuddingEveryday** [comment]
Tried using some different ports, still seems to produce the same thing.
